server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -727,6 +727,10 @@ public Map<String, UploadedSegmentMetadata> getSegmentsUploadedToRemoteStore() {
      * @throws IOException in case of I/O error while reading from / writing to remote segment store
      */
     public void deleteStaleSegments(int lastNMetadataFilesToKeep) throws IOException {
+        if (lastNMetadataFilesToKeep <= 0) {
+            logger.info("Stale segment deletion is disabled if cluster.remote_store.index.min.segment-metadata is set to < 1");
+            return;
+        }
```

Comment:
This would leave even the deleted indices data undeleted and leaked . A way to prevent would be to check this in `RemoteStoreRefreshListener` .

Reply:
Right, did not notice this. I will add a validator to not allow setting the value as 0